common/lib/xmodule/xmodule/course_module.py

```diff
@@ -1583,3 +1583,19 @@ def set_user_partitions_for_scheme(self, partitions, scheme):
             if p.scheme != scheme
         ]
         self.user_partitions = other_partitions + partitions  # pylint: disable=attribute-defined-outside-init
+
+    @property
+    def can_toggle_course_pacing(self):
+        """
+        Whether or not the course can be set to self-paced at this time.
+
+        Returns:
+          bool: False if the course is currently running, True otherwise.
+        """
+        now = datetime.now(UTC())
+        if self.start <= now:
+            if self.end is None:
+                return False
+            else:
+                return now >= self.end
+        return True
```

Comment:
cc: @marcotuts @mikigoyal 

This should prevent you from changing at any time after the course start date, rather than just when the course is running.